### PR TITLE
feat(web-app): add design kit library page 

### LINKED ISSUE #973: Library page for compatible design kits
Every library needs a `Design kits` page.

Redline: https://www.figma.com/file/K6HobXD4KXUIdkLrVjUzwK/LibraryDetail_CarbonReact?node-id=0%3A1

- [x] Body copy that says how these are "compatible" or "related" design kits that contain the design counterpart to coded assets in this library, with a link out to the design kits catalog to view all @aubrey-oneal 
- [x] Empty state for when there are no related design kits @kingtraceyj 
- [x] New resource card (MDX component for design kit resource card?) with data that comes from indexed design kits

image

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #973

#### Changelog

**New**

- add colored Axure icon
- add colored Invision icon
- add component prop to ResourceCard
- create design kit library page

**Changed**

- remove background from xd icon

#### Testing / reviewing

http://localhost:3000/libraries/carbon-react/latest/design-kits
http://localhost:3000/libraries/ibm-cloud-cognitive/latest/design-kits

All libraries have kits, below is a screen shot of the empty state. Or you can swap the logic on lines 98/99 in `design-kits.js `to test locally. We're holding on adding the contact card until the contact sections are added. Being tracked in #1044

**Screenshots for design review @kingtraceyj** 

screenshot

FireShot Capture 001 - IBM Products design kits - Carbon Design System - localhost

empty

### CODE REVIEW

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js [hunk 1]
```diff
@@ -4,13 +4,26 @@
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
-import { Column, Grid } from '@carbon/react'
+
+import { Column, Grid, Link as CarbonLink, Tag } from '@carbon/react'
+import { RulerAlt } from '@carbon/react/icons'
+import { AnchorLink, AnchorLinks } from '@carbon-platform/mdx-components'
+import { get, groupBy } from 'lodash'
```
Comment (francinelucca) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js:11]: can we use the new form of import here? I think it's this
```suggestion
import get from 'lodash/get'
import groupBy from 'lodash/groupBy'
```

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js [hunk 2]
```diff
@@ -24,7 +37,18 @@ const DesignKits = ({ libraryData, navData }) => {
 
   const pageHeader = pageHeaders?.library ?? {}
 
-  const { name } = libraryData.content
+  const { name, designKits } = libraryData.content
+
+  const groupedDesignKits = groupBy(
+    Object.entries(designKits).map((key) => ({ ...key[1] })),
+    (kit) => kit.tool
+  )
```
Comment (francinelucca) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js:45]:
```suggestion
  const groupedDesignKits = groupBy(
    Object.values(designKits),
    (kit) => kit.tool
  )
```
or if you need it to be spread:
```
 const groupedDesignKits = groupBy(
    Object.values(designKits).map((val) => ({ ...val })),
    (kit) => kit.tool
  )
```

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js [hunk 3]
```diff
@@ -47,13 +71,76 @@ const DesignKits = ({ libraryData, navData }) => {
...
+                  to get started. You can also view all design kits in the{' '}
+                  <Link href="/design-kits" passHref>
+                    <CarbonLink>catalog</CarbonLink>
+                  </Link>
+                  .
+                </PageDescription>
+                <AnchorLinks>
+                  {filteredDesignTools.length > 2 &&
+                    filteredDesignTools.map((item, i) => (
+                      <AnchorLink key={i}>{get(designTools, `[${item}].name`)}</AnchorLink>
```
Comment (francinelucca) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js:116]: I think we agreed not to use `lodash.get`? no real preference here 🤷‍♀️ 
```suggestion
                      <AnchorLink key={i}>{designTools?.[item]?.name}</AnchorLink>
```

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js [hunk 4]
```diff
@@ -47,13 +71,76 @@ const DesignKits = ({ libraryData, navData }) => {
...
+                    ))}
+                </AnchorLinks>
+                {filteredDesignTools.map((item, i) => (
+                  <>
+                    <H2 key={i}>{get(designTools, `[${item}].name`)}</H2>
+                    <CardGroup>
+                      {groupedDesignKits[`${item}`].map((kit, i) => (
+                        <Column lg={4} md={4} key={i}>
```
Comment (francinelucca) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js:124]: Small breakpoint doesn't seem to follow redline 
image

image

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js [hunk 5]
```diff
@@ -24,7 +37,18 @@ const DesignKits = ({ libraryData, navData }) => {
 
   const pageHeader = pageHeaders?.library ?? {}
 
-  const { name } = libraryData.content
+  const { name, designKits } = libraryData.content
+
+  const groupedDesignKits = groupBy(
```
Comment (francinelucca) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js:42]: this is triggering an error for. me if I go on a library that doesn't have a `designKits` key (not empty array, but the key is undefined, which will happen if it wasn't specified in the carbon.yml):
image

**Review by francinelucca — CHANGES_REQUESTED** [review]
Looking goood! 💃 
Noticed some buggies and added some code suggestions

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js (hunk 4 above)
Comment (alisonjoseph) [reply services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js]: oops! was missing sm={4} fixed

File: services/web-app/components/mdx-icon/mdx-icon.js [hunk 6]
```diff
@@ -51,7 +55,9 @@ const svgIcons = {
   adobe24: Svg24Adobe,
   axure24: Svg24Axure,
   adobe32: Svg32Adobe,
-  axure32: Svg32Axure
+  axure32: Svg32Axure,
+  xd: Svg32AdobeXd,
```
Comment (francinelucca) [inline services/web-app/components/mdx-icon/mdx-icon.js:59]: should this be `xd32` instead?
```suggestion
  xd32: Svg32AdobeXd,
```

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js [hunk 7]
```diff
@@ -47,13 +60,90 @@ const DesignKits = ({ libraryData, navData }) => {
...
+                    <CarbonLink>catalog</CarbonLink>
+                  </Link>
+                  .
+                </PageDescription>
+                <AnchorLinks>
+                  {filteredDesignTools.length > 2 &&
+                    filteredDesignTools.map((item, i) => (
+                      <AnchorLink key={i}>{designTools?.[item]?.name}</AnchorLink>
+                    ))}
+                </AnchorLinks>
```
Comment (francinelucca) [inline services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js:121]: is the spacing here correct? maybe the conditional for `.length >2` should be outisde of `<AnchorLinks>` so the extra spacing doesn't render?
image

**Review by francinelucca — CHANGES_REQUESTED** [review]
LGTM! just 2 qs

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js (hunk 7 above)
Comment (alisonjoseph) [reply services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js:121]: Oh yea, good catch!

**Review by aubrey-oneal — CHANGES_REQUESTED** [review]
Looks good! Just caught a small typo.

**Review by kingtraceyj — CHANGES_REQUESTED** [review]
One small update:

**hover state for xd card:**
Xd icon should have transparent background on hover
image

**Comment by alisonjoseph** [comment]
@kingtraceyj that should be fixed in the PR already, might need to rebuild the packages to see the updated icon (it won't get generated until build)

**Comment by alisonjoseph** [comment]
@aubrey-oneal fixed!

File: services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js [hunk 8]
```diff
@@ -47,13 +60,91 @@ const DesignKits = ({ libraryData, navData }) => {
...
+    )
+  }
+
+  const NoDesignKits = () => {
+    return (
+      <>
+        <PageDescription>This library contains no compatible design kits.</PageDescription>
+        <P>
+          It looks like there are no design kits compatible with this library. Explore the design
+          kit catalog to find more design kits or contact libray maintainers.
```
[the remark this answers is not in the document]
Comment (alisonjoseph) [reply services/web-app/pages/libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits.js]: fixed!

**Review by francinelucca — APPROVED** [review]
💃 nicee

**Review by andreancardona — APPROVED** [review]
🔥

**Review by aubrey-oneal — APPROVED** [review]

**Comment by jeanservaas** [comment]
@kingtraceyj @alisonjoseph just noticed this

do we not have the same icon as the design kit pictogram here

image

image

**Comment by alisonjoseph** [comment]
@jeanservaas I believe that Pictogram doesn't exist as an icon, so @kingtraceyj chose the ruler as the closest match. Let me know if I should switch it to something else.

**Review by kingtraceyj — APPROVED** [review]
looks good to go!

**Pull request merged**